File: 프로그래머스/2단계/네트워크.js
function solution(n, computers) {
  let check = [];
  let network = 0;

  for (let i = 0; i < computers.length; i++) {
    check[i] = false;
  }

  function dfs(index) {
    check[index] = true;

    for (let i = 0; i < computers.length; i++) {
      if (computers[index][i] == 1 && !check[i]) {
        dfs(i);
      }
    }
  }

  for (let i = 0; i < computers.length; i++) {
    if (!check[i]) {
      dfs(i);
      network++;
    }
  }

  return network;
}

console.log(
  solution(3, [
    [1, 1, 0],
    [1, 1, 0],
    [0, 0, 1],
  ])
);

File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
def dfs(visit, index, computers):
    visit[index] = 1
    
    for i in range(len(computers)):
        if visit[i] == 0 and computers[i][index] == 1:
            dfs(visit, i, computers)

def solution(n, computers):
    answer = 0
    visit = [0] * len(computers)
    
    for i in range(len(computers)):
        if visit[i] == 0:
            dfs(visit, i, computers)
            answer += 1
    return answer